gradoop-flink/src/main/java/org/gradoop/flink/model/api/operators/UnaryBaseGraphToGraphOperator.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright © 2014 - 2018 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.model.api.operators;
+
+import org.gradoop.flink.model.api.epgm.BaseGraph;
+
+/**
+ * Creates a graph of type {@link LG} based on one input graph of the same type.
+ *
+ * @param <LG> the type of the graph used as input and return value.
+ */
+public interface UnaryBaseGraphToGraphOperator<LG extends BaseGraph>
```

Comment:
Maybe rename to UnaryBaseGraphToBaseGraphOperator? Might be too long though.

Reply:
Maybe we need this naming in the future if we provide an operator that converts a BaseGraph to an _other_ BaseGraph.

Reply:
OK *thumbsup* ... if we will have an operator that converts a base graph to an other base graph in the future, you have to find a name for it XD